plugin_runner/plugin_runner.py

```diff
@@ -182,6 +189,38 @@ async def ReloadPlugins(
             yield ReloadPluginsResponse(success=True)
 
 
+def validate_effects(effects: list[Effect]) -> list[Effect]:
+    """Validates the effects based on predefined rules."""
+    autocomplete_effects_count = sum(
+        1 for effect in effects if effect.type == EffectType.AUTOCOMPLETE_SEARCH_RESULTS
+    )
+
+    if autocomplete_effects_count > 1:
+        log.warning("Only one AUTOCOMPLETE_SEARCH_RESULTS effect is allowed.")
+        return []
```

Comment:
What is the desired behavior here if they are also returning non-search-related effects as well? This seems like it will turn the entire plugin into a no-op, even the non-search-related effects.

That's one option. A few more:

* Log the warning as you are, and use the first search-related effect.
* Log the warning as you are, and just remove all search-related effects but keep the rest.

Reply:
opted for option 1.